volatility3/framework/plugins/windows/skeleton_key_check.py

```diff
@@ -0,0 +1,578 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+# This module attempts to locate skeleton-key like function hooks.
+# It does this by locating the CSystems array through a variety of methods,
+# and then validating the entry for RC4 HMAC (0x17 / 23)
+#
+# For a thorough walkthrough on how the R&D was performed to develop this plugin, 
+# please see our blogpost here:
+#
+# <insert blog URL once published>
+
+import logging, io
+
+from typing import Iterable, Tuple
+
+from volatility3.framework.symbols.windows import pdbutil
+from volatility3.framework import interfaces, symbols, exceptions
+from volatility3.framework import renderers, constants
+from volatility3.framework.layers import scanners
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.symbols import intermed
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import pslist, vadinfo
+
+from volatility3.framework.symbols.windows.extensions import pe
+
+try:
+    import capstone
+    has_capstone = True
+except ImportError:
+    has_capstone = False
+
+try:
+    import pefile
+    has_pefile = True
+except ImportError:
+    has_pefile = False
+
+vollog = logging.getLogger(__name__)
+
+class Skeleton_Key_Check(interfaces.plugins.PluginInterface):
+    """Lists process memory ranges that potentially contain injected code."""
```

Comment:
I updated the doc here to be correct. The blog post will replace the marker in the beginning comments once the code and blog post are live / MR accepted. It will have a ton of documentation as the data structures were all reverse engineered.